packages/flutter/lib/src/gestures/tap.dart

```diff
@@ -230,16 +233,7 @@ class TapGestureRecognizer extends PrimaryPointerGestureRecognizer {
   }
 
   void _checkUp() {
-    if (_wonArenaForPrimaryPointer && _finalPosition != null) {
```

Comment:
The other calls to _checkUp in this file do not need this check?

Reply:
No. At L210 `_wonArenaForPrimaryPointer ` was immediately set to true before calling `_checkUp`. At L177 `_checkUp` was in a `if(_wonArenaForPrimaryPointer)`.